tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersVariables.cs

```diff
@@ -0,0 +1,815 @@
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 68 } }", 771)]
+        [BuildersMQL("{ \"Root.Data\" : 25 }", 773)]
+        [BuildersMQL("{ \"Root.Left.Data\" : { \"$lte\" : 22 } }", 774)]
+        [BuildersMQL("{ \"Root.Data\" : { \"$ne\" : 20 } }", 776)]
+        [BuildersMQL("{ \"Age\" : { \"$lt\" : 21 } }", 777)]
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 68 } }", 779)]
+        public void Nested_conditional_expressions()
+        {
+            var w = Builders<Tree>.Filter.Eq(t => t.Root.Data, 25);
+            var x = Builders<Tree>.Filter.Lte(t => t.Root.Left.Data, 22);
+            var z = Builders<Tree>.Filter.Ne(t => t.Root.Data, 20);
+            var a = Builders<User>.Filter.Lt(u => u.Age, 21);
+            var b = Builders<User>.Filter.Gt(u => u.Height, 65);
+            var c = Builders<User>.Filter.Gt(u => u.Height, 68);
+            var d = Builders<User>.Filter.Gt(u => u.Height, 66);
+            var e = Builders<User>.Filter.Gt(u => u.Height, 67);
+
+            var y = w == Builders<Tree>.Filter.Eq(t => t.Root.Data, 26) ? z == Builders<Tree>.Filter.Eq(t => t.Root.Data, 36) ? e = a: d = e : b = c;
+
+            var finalResultW = w;
+            var finalResultX = x;
+            var finalResultY = y;
+            var finalResultZ = z;
+            var finalResultA = a;
+            var finalResultB = b;
+            var finalResultC = c;
+            var finalResultD = d;
+            var finalResultE = e;
+        }
+
+        [BuildersMQL("{ \"Root.Left.Data\" : { \"$lte\" : 22 } }", 794)]
+        [BuildersMQL("{ \"Root.Data\" : 25 }", 796)]
+        [BuildersMQL("{ \"Address\" : { \"$exists\" : true } }", 800)]
+        [BuildersMQL("{ \"Address\" : { \"$exists\" : false } }", 801)]
+        [BuildersMQL("{ \"Address\" : { \"$exists\" : true } }", 802)]
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 67 } }", 804)]
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 65 } }", 805)]
+        [BuildersMQL("{ \"Height\" : { \"$gt\" : 67 } }", 806)]
+        public void Member_access_and_element_access_should_not_be_displayed()
+        {
+            var x = Builders<Tree>.Filter.Lte(t => t.Root.Left.Data, 22);
+            var array = new FilterDefinition<Tree>[1];
+            array[0] = Builders<Tree>.Filter.Eq(t => t.Root.Data, 25);
+            var arrayRef = array;
+
+            var nestedFilter = new NestedFilter();
+            var PersonFilter = Builders<Person>.Filter.Exists(p => p.Address, true);
+            nestedFilter.PersonFilter = Builders<Person>.Filter.Exists(p => p.Address, false);
+            var finalResultPersonFilter = PersonFilter;
+
+            var UserFilter = Builders<User>.Filter.Gt(u => u.Height, 67);
+            nestedFilter.UserFilter = Builders<User>.Filter.Gt(u => u.Height, 65);
+            var finalResultUserFilter = UserFilter;
+        }
+
+        private FilterDefinition<T> GenerateBuildersExpression<T>(FilterDefinition<T> buildersExpression) => buildersExpression;
```

Comment:
Maybe we should also add cases referencing local class members (fields and methods).

Reply:
There is an issue with adding local class fields that are builders expressions. this is not related to vs37 at all. In TestCasesRunner, the following code snippet:         var diagnosticsAndMethodNodes = diagnostics
            .Where(d => DiagnosticRulesConstants.AllRules.Contains(d.Descriptor.Id))
            .Select(d =>
                (Diagnostic: d,
                 MethodNode: d.Location.SourceTree.GetRoot()
                 .FindNode(d.Location.SourceSpan)
                 .Ancestors()
                 .OfType<MethodDeclarationSyntax>()
                 .First()))
            .ToArray();
finds diagnostics that come from a Method. It's not part of this ticket but wondering if i can just change First() to FirstOrDefault() just to allow this testing. I changed it locally and observed that it worked well for local class fields.